File: advent_of_code/solutions/year2025/day02/python/solution_test.py
#!/usr/bin/env python3
"""Tests for Advent of Code 2025 Day 2"""

import json
from pathlib import Path

import pytest

from advent_of_code.solutions.year2025.day02.python.solution import part1, part2


@pytest.fixture
def input_text():
    return Path("advent_of_code/solutions/year2025/day02/input.txt").read_text()


@pytest.fixture
def answers():
    answers_path = Path("advent_of_code/solutions/year2025/day02/answers.json")
    if not answers_path.exists():
        pytest.skip("answers.json not found")
    return json.loads(answers_path.read_text())


def test_part1(input_text, answers):
    if "part1" not in answers or answers["part1"] is None:
        pytest.skip("part1 answer not yet verified")

    got = part1(input_text)
    want = int(answers["part1"])
    assert got == want, f"part1() = {got}, want {want}"


def test_part2(input_text, answers):
    if "part2" not in answers or answers["part2"] is None:
        pytest.skip("part2 answer not yet verified")

    got = part2(input_text)
    want = int(answers["part2"])
    assert got == want, f"part2() = {got}, want {want}"

File: projects/advent_of_code/solutions/year2015/day04/go/solution_test.go
package main

import (
	"encoding/json"
	"os"
	"testing"

	"github.com/jomcgi/homelab/projects/advent_of_code/pkg/aoc"
)

type Answers struct {
	Part1 *string `json:"part1,omitempty"`
	Part2 *string `json:"part2,omitempty"`
}

func loadAnswers(t *testing.T) *Answers {
	data, err := os.ReadFile("projects/advent_of_code/solutions/year2015/day04/answers.json")
	if err != nil {
		t.Skipf("answers.json not found: %v", err)
	}
	var answers Answers
	if err := json.Unmarshal(data, &answers); err != nil {
		t.Fatalf("failed to parse answers.json: %v", err)
	}
	return &answers
}

func TestPart1(t *testing.T) {
	answers := loadAnswers(t)
	if answers.Part1 == nil {
		t.Skip("part1 answer not yet verified")
	}

	input, err := aoc.ReadFile("projects/advent_of_code/solutions/year2015/day04/input.txt")
	if err != nil {
		t.Fatalf("failed to read input: %v", err)
	}

	got := part1(input)
	want := aoc.MustInt(*answers.Part1)
	if got != want {
		t.Errorf("part1() = %d, want %d", got, want)
	}
}

func TestPart2(t *testing.T) {
	answers := loadAnswers(t)
	if answers.Part2 == nil {
		t.Skip("part2 answer not yet verified")
	}

	input, err := aoc.ReadFile("projects/advent_of_code/solutions/year2015/day04/input.txt")
	if err != nil {
		t.Fatalf("failed to read input: %v", err)
	}

	got := part2(input)
	want := aoc.MustInt(*answers.Part2)
	if got != want {
		t.Errorf("part2() = %d, want %d", got, want)
	}
}
